### ISSUE TITLE
Websockets support?

### ISSUE THREAD

**Issue by tai41** [stagemonitor/stagemonitor-mailinglist#58]
Is there any plan to later support websockets protocol (in addition to just http)?
I've been trying to see more trace data on one of my servlets, but since websockets is a different protocol altogether, and the data flow is usually server-to-client (instead of the traditional client-to-server requests), I've been largely unsuccessful.

I tried calling "stagemonitor.init()" and adding the "@Traced" annotations, and I'm now able to see the method names in the metrics tab. However, I don't see any numbers on any of them aside from the "All" row. Is there any way to just see what methods get invoked in a class and possibly trace it (regardless of protocol)?

**Comment by felixbarny** [comment]
> Is there any plan to later support websockets protocol (in addition to just http)?

There are no concrete plans yet, but that should be possible. Which framework are you using for websockets? Plain websocket API?

> Is there any way to just see what methods get invoked in a class and possibly trace it (regardless of protocol)?

I guess you could see that in the call tree.

**Comment by tai41** [comment]
Yes, just the standard implementation for websockets (https://docs.oracle.com/javaee/7/api/javax/websocket/package-summary.html).

And regarding the call tree in the browser plugin, I see methods being invoked - but only those initiated by http requests. I was, however, able to see them through kibana as that seems to catch _all_ methods being invoked regardless of what initiated the call tree (i.e. the parent level call wasn't an http web request - but rather an independent thread). So that's all working now.

Thanks.
